packages/sonata-extra-bundle/Extension/AutoActionExtension.php

```diff
@@ -38,13 +38,23 @@ public function configureListFields(ListMapper $list): void
             }
         }
 
-        $list->add(
-            ListMapper::NAME_ACTIONS,
-            ListMapper::TYPE_ACTIONS,
-            [
-                'label' => 'Action',
-                'actions' => $this->actions,
-            ]
-        );
+        $routes = $admin->getRoutes();
+
+        foreach ($this->actions as $action => $options) {
```

Comment:
Use a temp array for the actions since the same extension instance will be use for multiple admin. One admin could have the action even if the other one doesn't